astroid/bases.py

```diff
@@ -293,6 +290,11 @@ class Instance(BaseInstance):
     # pylint: disable=unnecessary-lambda
     special_attributes = lazy_descriptor(lambda: objectmodel.InstanceModel())
 
+    def __init__(self, proxied: nodes.ClassDef) -> None:
+        if not isinstance(proxied, nodes.ClassDef):
+            raise TypeError
```

Comment:
```suggestion
```

That _should_ be unreachable 😄

Reply:
I don't follow -- I thought you asked for this guard? Otherwise the constructor is pointless.

Reply:
No, the constructor is helpful for type checkers. `mypy` doesn't recognise that `Instance._proxied: ClassDef` (just a couple lines above) means that for `Instance(x)` `x` needs to be `ClassDef`. By making the constructor explicit we will get type warnings from `mypy` (when we turn it on) if we create an `Instance` without making sure `x` is `ClassDef`.

Reply:
gotcha!